lib/Service/ApiService.php

```diff
@@ -97,6 +99,11 @@ public function create(?int $fileId = null, ?string $filePath = null, ?string $b
 
 			$this->sessionService->removeInactiveSessionsWithoutSteps($file->getId());
 			$document = $this->documentService->getDocument($file->getId());
+			$resetAfterUpdateTimestamp = $this->config->getValueInt(Application::APP_NAME, 'update_reset_before', 0);
+			if ($document && $document->getLastSavedVersionTime() < $resetAfterUpdateTimestamp) {
+				$this->documentService->resetDocument($document->getId());
+			}
```

Comment:
Will the `resetDocument` bump the value of `getLastSavedVersionTime`? Otherwise this will be run for every session created until the next (auto-)save.
In particular for folder descriptions this can take a loong time if they are never changed and people randomly create sessions when visiting the folder.
What about read only sessiions? Can they reset the document? Will that update the last saved version time?

Reply:
Ah... i see... `resetDocument` will delete the entry from the documents table. So `$document` will be empty for the next request coming in.

One thing that might still get in the way would be locks - `resetDocument` tries to unlock the file. I don't know what would happen if the file is not locked by text.